src/Map.ts

```diff
@@ -285,7 +285,7 @@ export class Map extends maplibregl.Map {
         // If the URL is present in the list of monitored style URL,
         // that means this AJAXError was about a style, and we want to fallback to
         // the default style
-        if (this.monitoredStyleUrls.has(clearnUrlStr)) {
+        if (this.monitoredStyleUrls && this.monitoredStyleUrls.has(clearnUrlStr)) {
```

Comment:
How about instead of doing this and:
```
    if (typeof this.monitoredStyleUrls === "undefined") {
      this.monitoredStyleUrls = new Set<string>();
    }
 ```
 
changing the:
```
private monitoredStyleUrls!: Set<string>;
```
to
```
private monitoredStyleUrls = new Set<string>();
```
?

Would it break anything?

Reply:
This is actually not possible because of the lifecycle of the class due to its inheritance. Here is the thing: whatever object attribute that is initialized outside of the constructor (like you suggest) is run after the `super()` call that lies in the constructor. We instantiate `this.monitoredStyleUrls` inside `monitorStyleUrl()` because `monitorStyleUrl()` is called by `setStyle()` and `setStyle()` is called by `super()`. This is so because SDK's `setStyle()` is overwriting Maplibre's `setStyle()` (but SDK's `setStyle()` calls Maplibre's `setStyle()` so we are good).

That means that `this.monitoredStyleUrls` is accessed before `super()` has finished to run and if instantiate the way you suggest (which is generally the right way) then its actually not going to be instantiated yet when the cascade inside `super()` needs it.